core/benchmark-tests.txt

```diff
@@ -0,0 +1,4 @@
+Benchmark                                                 Mode  Cnt      Score     Error   Units
+ScopeImplBenchmark.recordingBenchmark                    thrpt   10     74.819 ±   4.573  ops/ms
```

Comment:
Please check how other benchmarks are reported (we need to capture all data-points offered by JMH)

Reply:
Sorry, I don't really understand how exactly to do it. 

Would you mind taking it over? I am not convinced that this should be done by this PR, as the benchmark is already missing. The most problem is I don't have a clear picture of what you really want for benchmarking, and it should be much easier if your team just do it. 

LMK if you agree.

Reply:
It should be done by this PR. This is a critical library laying in the hot-path of execution of many services, hence we need to maintain the focus on its performance. 

Over the last year a lot of efforts have been put in to optimize, streamline, and make this library robust to serve the needs it was originally built for. Therefore, as a new contribution guideline we require any non-trivial change to adhere to the same basic principles of assuring library's correctness and performance.

Totally appreciate the amount of incremental effort to require to adhere to this heightened standards of every contribution, but unfortunately there's no other way to guarantee high-level of reliability and performance of the open-source library otherwise.

Reply:
> Please check how other benchmarks are reported

Can you give more details here? I did check but don't know how that works(though I see other reports have the details like gc time). 
For example another benchmark is just
`public class M3ReporterBenchmark extends AbstractReporterBenchmark`. So I don't know what am I missing here.

Reply:
@SokolAndrey can you help in here?